File: Easy/Python/1232. Check If It Is a Straight Line.py
'''
You are given an array coordinates, coordinates[i] = [x, y], where [x, y] represents the coordinate of a point. Check if these points make a straight line in the XY plane.
'''

class Solution:
    def checkStraightLine(self, coordinates: list[list[int]]) -> bool:
        if len(coordinates) <= 2:
            return True
        else:
            start_x = coordinates[0][0]
            start_y = coordinates [0][1]
            delta_y = coordinates[1][1] - start_y
            delta_x = coordinates[1][0] - start_x

            for x,y in coordinates[2:]:
                curr_delta_x = x - start_x
                curr_delta_y = y - start_y

                if curr_delta_y * delta_x != curr_delta_x * delta_y:
                    return False

            return True

File: Easy/Javascript/1232. Check If It Is a Straight Line.js
/**You are given an array coordinates, coordinates[i] = [x, y], 
 * where [x, y] represents the coordinate of a point. 
 * Check if these points make a straight line in the XY plane. 
 */


/**
 * @param {number[][]} coordinates
 * @return {boolean}
 */

var checkStraightLine = function (coordinates) {
    if (coordinates.length <= 2) {
        return true
    }

    let x0 = coordinates[0][0];
    let y0 = coordinates[0][1];
    let dx = coordinates[1][0] - x0;
    let dy = coordinates[1][1] - y0

    for (let coordinate of coordinates.slice(2)) {
        if ((coordinate[1] - y0) * dx != (coordinate[0] - x0) * dy) {
            return false
        }
    }

    return true
};

console.log(checkStraightLine([[0,0],[0,1],[0,-1]]))